src/framework/components/element/component.js

```diff
@@ -712,19 +714,35 @@ pc.extend(pc, function () {
         //   localAnchor, width, height, (local position is updated if anchors are split)
         // assumes these properties are up to date
         //   _margin
-        _calculateSize: function () {
+        _calculateSize: function (propagateCalculatedWidth, propagateCalculatedHeight) {
             // can't calculate if local anchors are wrong
             if (!this.entity._parent && !this.screen) return;
 
             this._calculateLocalAnchors();
 
-            var p = this.entity.getLocalPosition();
+            var newWidth = this._absRight - this._absLeft;
+            var newHeight = this._absTop - this._absBottom;
 
-            this._setWidth(this._absRight - this._absLeft);
-            this._setHeight(this._absTop - this._absBottom);
+            // TODO This change is significant - needs discussion.
+            //      For context: without this change, any modifications to the margins done when
+            //      calculatedWidth/Height is updated will eventually end up being propagated to
+            //      the regular width/height. We want to avoid this except in the special case
+            //      where the user has changed the anchors from being split to unsplit.
```

Comment:
I added this TODO to draw some attention to this section during the review, as it's significant. I'll remove it after discussion.

Reply:
I'm fine with it if it doesn't break stuff

Reply:
agreed